File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.java
class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
        var map1 = new HashMap<Integer, Integer>();
        var map2 = new HashMap<Integer, Integer>();
        for(int n: nums1)
            map1.put(n, map1.getOrDefault(n, 0)+1);
        for(int n: nums2)
            map2.put(n, map2.getOrDefault(n, 0)+1);
        
        var list = new ArrayList<Integer>();
        for(int k: map1.keySet()){
            for(int i=0; i<Math.min(map1.get(k), map2.getOrDefault(k, 0)); ++i)
                list.add(k);
        }
        
        int[] ans = new int[list.size()];
        for(int i=0; i<list.size(); ++i) ans[i] = list.get(i);
        return ans;
    }
}

File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        c1, c2 = Counter(nums1), Counter(nums2)
        ans = []
        for k,v in c1.items():
            for i in range(min(v, c2.get(k, 0))):
                ans.append(k)
        return ans